### ISSUE TITLE
Unable to use a tag that starts with aws: in a datasource filter

### ISSUE THREAD

**Issue by Dchamard** [hashicorp/terraform#11838]
### Terraform Version
0.8.4

### Affected Resource(s)

- aws_security_group (datasource)

It is propably affecting most of the aws datasources as well.

### Terraform Configuration Files
```hcl
data "aws_security_group" "beanstalk_sg" {
  count = "${var.beanstalk_sg_manage_enabled}"
  tags = {
     "aws:cloudformation:logical-id"     =  "AWSEBLoadBalancerSecurityGroup"
     "Name"                               = "${var.stack_name}"
     "service"                           =  "${var.stack_name}-beanstalk"
  }
}
```

### Debug Output
2017/02/09 17:14:25 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/09 17:14:25 [DEBUG] Matching ^aws:* with Name
2017/02/09 17:14:25 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/09 17:14:25 [DEBUG] Matching ^aws:* with service
2017/02/09 17:14:25 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/09 17:14:25 [DEBUG] Matching ^aws:* with aws:cloudformation:logical-id
**2017/02/09 17:14:25 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/09 17:14:25 [DEBUG] Found AWS specific tag aws:cloudformation:logical-id (val: AWSEBLoadBalancerSecurityGroup), ignoring.**
2017/02/09 17:14:25 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/09 17:14:25 [DEBUG] Describe Security Groups {

### Expected Behavior
I should have been abled to use this tag to find the proper resource for my data source

### Actual Behavior
It is ignoring the tag because it starts with aws:
This was introduced in the following issue.
https://github.com/hashicorp/terraform/issues/4513

I understand that those tags cannot be modifed because its used internally by aws but we should be able to use them as data source to find resources.

The data source is using the same  'tagsFromMap' generic method that calls the 'tagIgnored' method that will filter tags that start with aws:

### References
https://github.com/hashicorp/terraform/issues/4513

**Comment by ericwestfall** [comment]
I have submitted a PR that provides the ability for data sources to opt-out of the internal AWS tag filtering. I've confirmed the bugfix resolves your issue.

**Sample Terraform**
```
data "aws_security_group" "selected" {
  tags = {
     "aws:cloudformation:logical-id"     =  "WebServerSecurityGroup"
  }
}

output "sg" {
   value = "${data.aws_security_group.selected.name}"
}
```

**Fails as expected with Terraform v0.8.6**
```
[...]
data.aws_security_group.selected: Refreshing state...
2017/02/11 17:58:39 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/11 17:58:39 [DEBUG] Matching ^aws:* with aws:cloudformation:logical-id
2017/02/11 17:58:39 [DEBUG] plugin: terraform: aws-provider (internal) 2017/02/11 17:58:39 [DEBUG] Found AWS specific tag aws:cloudformation:logical-id (val: WebServerSecurityGroup), ignoring.
[...]
Error refreshing state: 1 error(s) occurred:

* data.aws_security_group.selected: multiple Security Groups matched; use additional constraints to reduce matches to a single Security Group
```

Works with development build that includes this fix:
```
LNGDAYM-4172503:~ westfaex$ /Users/westfaex/code/go/src/github.com/hashicorp/terraform/bin/terraform --version
Terraform v0.9.0-dev (f1732db2c1e2583841fe090b5eb91856ef831548)

LNGDAYM-4172503:~ westfaex$ /Users/westfaex/code/go/src/github.com/hashicorp/terraform/bin/terraform apply
data.aws_security_group.selected: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

sg = TestStack-WebServerSecurityGroup-WSPFNIQDBTPW
```

**Comment by dpetzel** [comment]
I ran into this as well tonight, if it helps I worked around this by using the `data "aws_cloudformation_stack"` data source and then reading the outputs I need from the cloudformation stack. Not suggesting this shouldn't get addressed just wanted to pass along a potential workaround in case it helps anyone else.

**Comment by ericwestfall** [comment]
I've closed PR #11885 and created a new PR #14116 that implements a fix for this in a much cleaner way. The new PR allows a data source to decide if they want to provide a way to opt-out of AWS reserved tag filtering. 

If a data source chooses to implement this new functionality, the end-user can control the reserved tag filtering behavior by setting an argument `filter_reserved_tags`. This flag defaults to `false` meaning the data source will allow referencing AWS reserved tags. 

The PR updates the following data sources to implement this new functionality:
- `aws_route53_zone`
- `aws_route_table`
- `aws_security_group`
- `aws_subnet`
- `aws_vpc`
- `aws_vpc_peering_connection`
- `aws_vpn_gateway`
